### PR TITLE
apply (most of) readability-qualified-auto changes

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/peer_list.cpp [hunk 1]
```diff
@@ -98,15 +98,15 @@ namespace libtorrent {
 
 	void peer_list::clear()
 	{
-		for (auto const p : m_peers)
+		for (auto* p : m_peers)
```
Comment (FranciscoPombal) [inline src/peer_list.cpp:101]: Why not `for (auto* const p : m_peers)` in these 2 cases?

Comment (arvidn) [reply src/peer_list.cpp:101]: it didn't seem all that important to make `p` const, since it's scope is very limited

Comment (FranciscoPombal) [reply src/peer_list.cpp:101]: Well, I don't really have any arguments against the change besides "moar `const` moar better", so yeah.

**Comment by arvidn** [comment]
I guess clang-tidy 7 doesn't have this checker either

**Comment by FranciscoPombal** [comment]
https://github.com/arvidn/libtorrent/pull/4790#issuecomment-649132669

> @arvidn You can check all supported checks with `clang-tidy -checks="*" --explain-config`
>
>Looks like for this one at least `clang-tidy` 8 is required. For `readability-qualified-auto` in https://github.com/arvidn/libtorrent/pull/4789, the minimum requirement is version 10.

`clang-tidy-10` is only available on Ubuntu >= 20.04. So for this one, either wait until Travis supports 20.04 images, or perhaps consider using LLVM's official apt repository: https://apt.llvm.org/

**Pull request merged**